cf-revision-manager.php

```diff
@@ -2,200 +2,321 @@
 /*
 Plugin Name: CF Revision Manager
 Plugin URI: http://crowdfavorite.com
-Description: Revision management functionality so that plugins can add metadata to revisions as well as restore that metadata from revisions
-Version: 1.0.2
+Description: Revision management functionality so that plugins can add metadata to revisions as well as restore that metadata from revisions.
+Version: 1.0.1
 Author: Crowd Favorite
-Author URI: http://crowdfavorite.com 
+Author URI: https://crowdfavorite.com/
+License: GPL2
+License URI: https://www.gnu.org/licenses/gpl-2.0.html
+Domain Path: /languages
+Text Domain: cfrm
+ */
+
+// Exit if already accessed
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+* Fetch global variables declared for use anywhere in the plugin
+*
+* @since 1.0.1
+*
+* @return mixed Returns the global value depending on the key being provided
 */
-if (!class_exists('cf_revisions')) {
-	
-	define('CF_REVISIONS_DEBUG', false);
-	
-	function cfr_register_metadata($postmeta_key, $display_func = '') {
-		static $cfr;
-		if (empty($cfr)) {
-			$cfr = cf_revisions::get_instance();
-		} 
-		return $cfr->register($postmeta_key, $display_func);
+function cfrm_get_global( $key ) {
+	$cfrm                   = array();
+	$cfrm['plugin_file']    = __FILE__;
+	$cfrm['basename']       = plugin_basename( __FILE__ );
+	$cfrm['directory_path'] = plugin_dir_path( __FILE__ );
+	$cfrm['directory_url']  = plugin_dir_url( __FILE__ );
+	return array_key_exists( $key, $cfrm ) ? $cfrm[$key] : '';
+}
+
+/**
+ * Register js and css file.
+ *
+ * @since 1.0.1
+ */
+function cfrm_enqueue_js() {
+	wp_register_script( 'cfrm-script', cfrm_get_global( 'directory_url' ) . 'cfrm-script.js', array( 'jquery' ) );
+	wp_register_style( 'cfrm-style', cfrm_get_global( 'directory_url' ) . 'cfrm-style.css' );
+}
+add_action( 'admin_enqueue_scripts', 'cfrm_enqueue_js' );
+
+/**
+ * Enqueue js and css file in revision page.
+ *
+ * @since  1.0.1
+ * @return void
+ */
+function cfrm_admin_head() {
+	if ( ! empty( $_GET['page'] ) && $_GET['page'] == basename( __FILE__ ) ) {
```

Comment:
If these scripts should only be enqueued on this plugins settings page, we can call `wp_enqueue_*` within the function that generates the settings page output, rather than hooking to admin_head and then checking a condition.